### PR TITLE
test: add unauthenticated user tests

### PR DESCRIPTION
**Pull request by kpark277**
Description of changes:
* Added unauthenticated user tests for all APIs
* Added missing negative tests

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/integration-tests/tests/isolated/projects/getDataSetForProject.test.ts [hunk 1]
```diff
@@ -84,5 +95,63 @@ describe('get dataset for project tests', () => {
...
+        dataSet1Id = dataSet1.id;
+      });
+
+      test('PA without access to dataset cannot get', async () => {
+        try {
+          await pa1Session.resources.projects.project(project2Id).dataSets().dataset(dataSet1Id).get();
+        } catch (error) {
+          checkHttpError(
+            error,
+            new HttpError(403, {
```
Comment (SanketD92) [inline solutions/swb-reference/integration-tests/tests/isolated/projects/getDataSetForProject.test.ts:124]: nit: declare up top

**Review by SanketD92 — APPROVED** [review]
Looks good overall

**Review by SanketD92 — APPROVED** [review]

File: solutions/swb-reference/integration-tests/tests/isolated/projects/listUsersForProject.test.ts [hunk 2]
```diff
@@ -173,16 +173,11 @@ describe('list users for project tests', () => {
       }
     });
 
-    test('ensure researcher cannot list users for projects in a different project', async () => {
+    test('ensure unauthenticated user cannot list users for projects', async () => {
```
Comment (aws-tyler) [inline solutions/swb-reference/integration-tests/tests/isolated/projects/listUsersForProject.test.ts:176]: is this replacing the researcher test completely?

File: solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts [hunk 3]
```diff
@@ -188,4 +190,57 @@ describe('list environment type configs associated to project', () => {
       });
     });
   });
+
```
Comment (aws-tyler) [inline solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts:193]: is there a test somewhere checking that ITAdmins can access the data?

**Review by aws-tyler — APPROVED** [review]

File: solutions/swb-reference/integration-tests/tests/isolated/projects/listUsersForProject.test.ts (hunk 2 above)
Comment (kpark277) [reply solutions/swb-reference/integration-tests/tests/isolated/projects/listUsersForProject.test.ts:176]: There was already a test for researcher on line 120.

File: solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts (hunk 3 above)
Comment (kpark277) [reply solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts:193]: Will address this in the multistep in a follow up PR.

Comment (kpark277) [reply solutions/swb-reference/integration-tests/tests/isolated/projectEnvTypeConfigs/list.test.ts:193]: Actually this already exists in multistep/et-etc.test.ts

**Pull request merged**